assembly/src/assembler/mod.rs

```diff
@@ -467,12 +467,13 @@ fn combine_blocks(mut blocks: Vec<CodeBlock>) -> CodeBlock {
     while blocks.len() > 1 {
         let last_block = if blocks.len() % 2 == 0 { None } else { blocks.pop() };
 
-        let mut grouped_blocks = Vec::new();
-        core::mem::swap(&mut blocks, &mut grouped_blocks);
-        let mut grouped_blocks = group_vector_elements::<CodeBlock, 2>(grouped_blocks);
-        grouped_blocks.drain(0..).for_each(|pair| {
-            blocks.push(CodeBlock::new_join(pair));
-        });
+        let mut source_blocks = Vec::new();
```

Comment:
Took me a couple times to make sure this was doing the right thing 😅, but it checks out. Definitely would be improved a ton with the ability to use an approach based on the unstable array/slice features, but it is what it is